magpie/cli/send_email.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Sends email notification using SMTP connection defined from configuration.
+
+Useful for validation of SMTP settings retrieved from an INI file or debugging the rendered email contents.
+"""
+import argparse
+import inspect
+from typing import TYPE_CHECKING
+
+from mako.template import Template
+
+from magpie.api.notifications import DEFAULT_TEMPLATE_MAPPING, get_email_template, send_email
+from magpie.cli.utils import make_logging_options, setup_logger_from_options
+from magpie.utils import get_logger, get_settings_from_config_ini
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, List, Optional, Sequence
+
+    from magpie.typedefs import Str
+    UserConfig = List[Dict[Str, Str]]
+
+LOGGER = get_logger(__name__,
+                    message_format="%(asctime)s - %(levelname)s - %(message)s",
+                    datetime_format="%d-%b-%y %H:%M:%S", force_stdout=False)
+
+
+class EmailTemplateChoiceFormatter(argparse.HelpFormatter):
+    def _format_action(self, action):
+        """
+        Override the returned help message with available options and shortcuts for email template selection.
+        """
+        text = super(EmailTemplateChoiceFormatter, self)._format_action(action)  # noqa: W0212
+        if action.dest != "template":
+            return text
+        self._indent()
+        indent = " " * self._current_indent
+        choices = action.choices.items()  # noqa
+        choices_str = ["{}{}) {}".format(indent, i, tmpl) for i, tmpl in choices if isinstance(i, int)]
+        text += "\n{}Choices are:\n{}".format(indent, "\n".join(choices_str))
+        self._dedent()
+        return text
+
+
+def make_parser():
+    # type: () -> argparse.ArgumentParser
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=EmailTemplateChoiceFormatter)
+    email_opts = parser.add_argument_group(title="Email Arguments",
+                                           description="Operation arguments to define email to be sent.")
+    email_opts.add_argument("-e", "--email", required=True, help="Address where to sent the test email.")
+    email_choices = ["test"] + list(DEFAULT_TEMPLATE_MAPPING)
+    email_choices_map = {i: tmpl for i, tmpl in enumerate(email_choices)}
+    email_choices_map.update({str(i): tmpl for i, tmpl in list(email_choices_map.items())})
+    email_choices_map.update({tmpl: tmpl for tmpl in email_choices})
```

Comment:
Indeed comment will help. 
Allows to map `"0" -> "test"` (key as string), `0 -> "test"` (key as int) and literal `"test" -> "test"` to corresponding templates. 
Mostly convenient for the long template names annoying to type, which are easier to select with just the index.
str/int mapping is needed for parsing since int as commandlien arg input becomes a string.